ocs_ci/utility/kms.py

```diff
@@ -761,36 +764,304 @@ def create_vault_csi_kms_token(
         self.create_resource(csi_kms_token, prefix="csikmstoken")
 
     def create_vault_csi_kms_connection_details(
-        self, kv_version, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        self,
+        kv_version,
+        kms_type=constants.VAULT_TOKEN,
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
     ):
         """
         Create vault specific csi kms connection details
         configmap resource
 
         """
+
         csi_kms_conn_details = templating.load_yaml(
             constants.EXTERNAL_VAULT_CSI_KMS_CONNECTION_DETAILS
         )
-        conn_str = csi_kms_conn_details["data"]["1-vault"]
-        buf = json.loads(conn_str)
-        buf["VAULT_ADDR"] = f"https://{self.vault_server}:{self.port}"
-        buf["VAULT_BACKEND_PATH"] = self.vault_backend_path
-        buf["VAULT_CACERT"] = get_default_if_keyval_empty(
-            config.ENV_DATA, "VAULT_CACERT", defaults.VAULT_DEFAULT_CA_CERT
-        )
-        buf["VAULT_NAMESPACE"] = self.vault_namespace
-        buf["VAULT_TOKEN_NAME"] = get_default_if_keyval_empty(
-            config.ENV_DATA, "VAULT_TOKEN_NAME", constants.EXTERNAL_VAULT_CSI_KMS_TOKEN
-        )
-        if kv_version == "v1":
-            buf["VAULT_BACKEND"] = "kv"
+        if kms_type == constants.VAULT_TOKEN:
```

Comment:
Not must but I feel nice to have: You can make these if and else part into seperate functions itself, since it has atleast 20 lines of code with specifics.